packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -78,26 +81,56 @@ export const styles = css`
     }
 
     .control:hover {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: none;
+        transition: box-shadow ${mediumDelay};
     }
 
     .control${focusVisible} {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: ${borderWidth} solid ${borderHoverColor};
-        outline-offset: -4px;
     }
 
     .control:active {
         box-shadow: none;
         outline: none;
+        transition: outline ${mediumDelay};
+        transition: box-shadow ${mediumDelay};
     }
 
     :host([disabled]) .control {
         box-shadow: none;
         outline: none;
     }
 
+    .control::before {
+        content: '';
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        pointer-events: none;
+        box-sizing: border-box;
+        outline: none;
+        background-clip: content-box;
+        z-index: -1;
+    }
+
+    .control${focusVisible}::before {
+        outline: ${borderWidth} solid ${borderHoverColor};
```

Comment:
The focus animations look very inconsistent to me.

For comparison, the hover animation:

![hoveranimation](https://github.com/ni/nimble/assets/1588923/1b28a70b-f6ef-4a26-bd9e-de9e49d74dcf)

the outer border appears to grow inward to two pixels over time.

there is no inner border

For the focus animation:

![focusanimation](https://github.com/ni/nimble/assets/1588923/e5cc2a7c-40fd-4e55-b92c-9b8d05e82afa)

The outer border does not seem to grow inward, it seems to pop-in instantly

The inner border seems to grow outward in size and it looks decoupled from the outer border animation. 

I can't find the reference animation so I can definitely be wrong but it seems overly complex. I would imagine it should feel like one single animation is happening where the outer border grows inward and the inner border either fades in or grows inward during the same time.

Reply:
This should be fixed in my newest commit, I slowed this video (below) down and it looks like the inner border is growing inward now.

https://github.com/ni/nimble/assets/103057880/57011312-428c-4bac-97ba-08b31981ed49